cpp/benchmarks/io/parquet/parquet_reader_input.cpp

```diff
@@ -231,6 +231,35 @@ void BM_parquet_read_chunks(nvbench::state& state, nvbench::type_list<nvbench::e
   state.add_buffer_size(source_sink.size(), "encoded_file_size", "encoded_file_size");
 }
 
+template <data_type DataType>
+void BM_parquet_read_wide_tables_mixed(nvbench::state& state,
+                                       nvbench::type_list<nvbench::enum_type<DataType>> type_list)
+{
+  auto const d_type = get_type_or_group(static_cast<int32_t>(DataType));
+
+  auto const n_col       = static_cast<cudf::size_type>(state.get_int64("num_cols"));
+  auto const num_rows    = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const cardinality = static_cast<cudf::size_type>(state.get_int64("cardinality"));
+  auto const run_length  = static_cast<cudf::size_type>(state.get_int64("run_length"));
+  auto const source_type = io_type::FILEPATH;
+  cuio_source_sink_pair source_sink(source_type);
+
+  {
+    auto const tbl = create_random_table(
+      cycle_dtypes(d_type, n_col),
+      row_count{num_rows},
+      data_profile_builder().cardinality(cardinality).avg_run_length(run_length));
+    auto const view = tbl->view();
+
+    cudf::io::parquet_writer_options write_opts =
+      cudf::io::parquet_writer_options::builder(source_sink.make_sink_info(), view)
+        .compression(cudf::io::compression_type::NONE);
+    cudf::io::write_parquet(write_opts);
+  }
+
+  parquet_read_common(num_rows, n_col, source_sink, state);
```

Comment:
parquet_read_common assumes data of size data_size. We need to add a parameter to parquet_read_common to reuse it here.